Valet/VALValet.m

```diff
@@ -237,37 +237,35 @@ + (nullable NSString *)_sharedAccessGroupPrefix;
 
 #pragma mark - Initialization
 
-- (nullable instancetype)initWithIdentifier:(nonnull NSString *)identifier accessibility:(VALAccessibility)accessibility;
+- (nonnull instancetype)initWithIdentifier:(nonnull NSString *)identifier accessibility:(VALAccessibility)accessibility;
 {
     VALCheckCondition(identifier.length > 0, nil, @"Valet requires an identifier");
     VALCheckCondition(accessibility > 0, nil, @"Valet requires a valid accessibility setting");
     
     self = [super init];
-    if (self != nil) {
-        _baseQuery = [[self class] mutableBaseQueryWithIdentifier:identifier accessibility:accessibility initializer:_cmd];
-        _identifier = [identifier copy];
-        _sharedAcrossApplications = NO;
-        _accessibility = accessibility;
-        _lockForSetAndRemoveOperations = [NSLock new];
-    }
+
+    _baseQuery = [[self class] mutableBaseQueryWithIdentifier:identifier accessibility:accessibility initializer:_cmd];
+    _identifier = [identifier copy];
+    _sharedAcrossApplications = NO;
+    _accessibility = accessibility;
+    _lockForSetAndRemoveOperations = [NSLock new];
     
     return [[self class] sharedValetForValet:self];
 }
 
-- (nullable instancetype)initWithSharedAccessGroupIdentifier:(nonnull NSString *)sharedAccessGroupIdentifier accessibility:(VALAccessibility)accessibility;
+- (nonnull instancetype)initWithSharedAccessGroupIdentifier:(nonnull NSString *)sharedAccessGroupIdentifier accessibility:(VALAccessibility)accessibility;
```

Comment:
I think this brings up a question regarding the philosophy behind the usage of `VALCheckCondition` here. As-is, it feels like the safety of returning nil in the case of an outright API misuse is not enough to justify the complexity of all consumers of the framework having to handle the potential of a Valet initialization being failable (returning nil). A consumer passing an empty string for a `SharedAccessGroupIdentifier` or a `0` value for `VALAccessibility` are both programmer error, not a runtime error that should be caught and handled appropriately. As such, it seems like these should be `@thrown` `NSInternalInconsistencyException`, rather than early nil returns.